dinersclub/fake.go

```diff
@@ -18,11 +19,16 @@ func NewFakeProvider() (Provider, error) {
 func (p *FakeProvider) GetUserFromPaymentEvent(event *PaymentEvent) (*User, error) {
 	if event.Pageid == "page" || event.Pageid == "935593143497601" {
 		return &User{Id:"test-id"}, nil
+	} else if event.Pageid == "test-auth-page" {
```

Comment:
This feels circular to me. I'm too tired to figure out what actually needs to be done, but I feel like we've added a lot of logic for a test that mostly tests the logic we added. 

Fake is essentially a mock, so maybe it's ok, but this feels like what we want is a mock.

Reply:
I see what you mean. Just pushed a new commit, https://github.com/vlab-research/fly/pull/42/commits/4225846c02da510d460eceaf709f956f95e39bfa, let me know what do you think.